src/assertions/threshold.jl

```diff
@@ -0,0 +1,38 @@
+"""
+    Threshold(ratio; weights=nothing)
+
+Assert that the ratio of missing values in the provided dataset does not exceed to specified ratio.
+If a weights array is provided then the ratio will be calculated as the
+`sum(weights[ismissing.(data)]) / sum(weights)`
+"""
+struct Threshold <: Assertion
+    ratio::Float64
+    weights::Union{AbstractVector, Nothing}
+end
+
+Threshold(ratio::Float64; weights=nothing) = Threshold(ratio, weights)
```

Comment:
The constructor will automatically call `convert`
ands I see no reason to disallow `Threshold(0.5f)`
or `Threadhold(0)` (though we might want a different `Assertion` type for that)

```suggestion
Threshold(ratio; weights=nothing) = Threshold(ratio, weights)
```

```suggestion
Threshold(ratio::Real; weights=nothing) = Threshold(ratio, weights)
```

Reply:
Turns out I made this a keyword in a later commit and didn't restrict it to `Float64` then.